owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt

```diff
@@ -1060,53 +1062,115 @@ class FileDisplayActivity : FileActivity(),
+                            repeat(data.asReversed().size) {
+                                replace.add(false)
+                                pos++
+                                if (pos == data.size) {
+                                    launchAction(uiResult.data, replace)
+                                }
+                            }
+                            dismissAllOpenDialogs()
+                        } else {
+                            replace.add(false)
+                            pos++
+                            if (pos == data.size) {
+                                launchAction(uiResult.data!!, replace)
+                            }
+                            customDialog.dismiss()
                         }
                     }
-                    .setNegativeButton(R.string.conflict_replace) { _, _ ->
-                        replace.add(true)
-                        pos++
-                        if (pos == data.size) {
-                            launchAction(uiResult.data, replace)
+
+                    override fun onSkipButtonClick() {
+                        if (customDialog.isCheckBoxChecked) {
+                            repeat(data.asReversed().size) {
+                                replace.add(null)
+                                pos++
+                                if (pos == data.size) {
+                                    launchAction(uiResult.data!!, replace)
+                                }
+                            }
+                            dismissAllOpenDialogs()
+                        } else {
+                            replace.add(null)
+                            pos++
+                            if (pos == data.size) {
+                                launchAction(uiResult.data!!, replace)
+                            }
+                            customDialog.dismiss()
                         }
                     }
-                    .setPositiveButton(R.string.conflict_keep_both) { _, _ ->
-                        replace.add(false)
-                        pos++
-                        if (pos == data.size) {
-                            launchAction(uiResult.data, replace)
+
+                    override fun onReplaceButtonClick() {
+                        if (customDialog.isCheckBoxChecked) {
+                            repeat(data.asReversed().size) {
+                                replace.add(true)
+                                pos++
+                                if (pos == data.size) {
+                                    launchAction(uiResult.data!!, replace)
+                                }
+                            }
+                            dismissAllOpenDialogs()
+                        } else {
+                            replace.add(true)
+                            pos++
+                            if (pos == data.size) {
+                                launchAction(uiResult.data!!, replace)
```

Comment:
Why we need the '!!' here, in previous versión we didn't use it.

Reply:
Done